src/main/java/com/ureca/uhyu/domain/auth/jwt/JwtAuthenticationFilter.java

```diff
@@ -1,4 +1,146 @@
 package com.ureca.uhyu.domain.auth.jwt;
 
-public class JwtAuthenticationFilter {
+import com.ureca.uhyu.domain.user.enums.UserRole;
+import com.ureca.uhyu.domain.user.repository.UserRepository;
+import com.ureca.uhyu.global.config.PermitAllURI;
+import io.jsonwebtoken.ExpiredJwtException;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.List;
+
+// jwt access token 에 대한 인가 확인 필터
+@Slf4j
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    @Value("${jwt.access-token-expiration-time}")
+    private long ACCESS_TOKEN_EXP;
+
+    private final JwtTokenProvider jwtTokenProvider;
+
+    // spring security의 인증 필터와 맞춰 줘야 함
+    @Override
+    protected boolean shouldNotFilter(HttpServletRequest request) {
+        String uri = request.getRequestURI();
+        return PermitAllURI.isPermit(uri);
+    }
+    
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
+                                    FilterChain filterChain) throws ServletException, IOException {
+
+        try {
+            String accessToken = extractAccessTokenFromCookie(request);
+
+            if (accessToken != null && jwtTokenProvider.validateToken(accessToken)) {
+
+                log.info("❗ access token 유효");
+
+                String userId = jwtTokenProvider.getUserIdFromToken(accessToken).toString();
+                String role = jwtTokenProvider.getRoleFromToken(accessToken);
+
+                if (role == null) {
+                    log.error("❌ access token에서 role을 추출하지 못했습니다. token: {}", accessToken);
+                    response.setStatus(HttpServletResponse.SC_UNAUTHORIZED); // 인증 실패 : 401에러
+                    return; // 필터
+                }
```

Comment:
```suggestion
if (role == null) {
    throw new InvalidTokenException("access token에서 role을 추출할 수 없습니다.");
}            
```
으로 바꿔서 예외로 처리하는게 좋지 않을까요? Role이 없는 Token이 들어오는건  잘못된 토큰인 상황이니까 예외로 처리하는게 맞다고 생각합니다.